libs/designsystem/src/lib/components/modal-experimental/footer/footer.component.spec.ts

```diff
@@ -0,0 +1,175 @@
+    expect(spectator.component).toBeTruthy();
+    ionFooterElement = spectator.query('ion-footer');
+    expect(ionFooterElement).toHaveComputedStyle({
+      'padding-left': BASE_PADDING_HORIZONTAL_PX,
+      'padding-right': BASE_PADDING_HORIZONTAL_PX,
+      'padding-top': BASE_PADDING_VERTICAL_PX,
+      'padding-bottom': BASE_PADDING_VERTICAL_PX,
+    });
+  });
+
+  describe('Set bottom padding', () => {
+    beforeEach(() => {
+      spectator = createHost(`<kirby-modal-footer-experimental></kirby-modal-footer-experimental>`);
+      modalFooterElement = spectator.element;
+      ionFooterElement = spectator.query('ion-footer');
+    });
+
+    it('when --kirby-safe-area-bottom is not set', () => {
+      expect(ionFooterElement).toHaveComputedStyle({ 'padding-bottom': BASE_PADDING_VERTICAL_PX });
+    });
+
+    it('when --kirby-safe-area-bottom is set', () => {
+      setSafeAreaBottom();
+      expect(ionFooterElement).toHaveComputedStyle({ 'padding-bottom': BASE_PADDING_VERTICAL_PX });
+    });
+
+    describe('on small screens', () => {
+      beforeAll(async () => {
+        await TestHelper.resizeTestWindow(TestHelper.screensize.phone);
+      });
+
+      afterAll(() => {
+        TestHelper.resetTestWindow();
+      });
+
+      it('when --kirby-safe-area-bottom is not set', () => {
+        expect(ionFooterElement).toHaveComputedStyle({
+          'padding-bottom': BASE_PADDING_VERTICAL_PX,
+        });
+      });
+
+      it('when --kirby-safe-area-bottom is set', () => {
+        setSafeAreaBottom();
+        const expected = BASE_PADDING_VERTICAL + SAFE_AREA_BOTTOM + 'px';
+        expect(ionFooterElement).toHaveComputedStyle({ 'padding-bottom': expected });
+      });
+    });
+  });
+
+  describe('Snap to keyboard', () => {
+    beforeEach(() => {
+      spectator = createHost(
+        `<kirby-modal-footer-experimental [snapToKeyboard]="snapToKeyboard"></kirby-modal-footer-experimental>`
+      );
+      modalFooterElement = spectator.element;
+      ionFooterElement = spectator.query('ion-footer');
+    });
+
+    describe('when snapToKeyboard is true', () => {
+      beforeEach(() => {
+        spectator.setHostInput('snapToKeyboard', true);
+      });
+
+      it('should follow the keyboard up', () => {
+        keyboardSlideIn();
+        expect(ionFooterElement).toHaveComputedStyle({
+          transform: TRANSFORM_PUSHED_BY_KEYBOARD,
+        });
+      });
+
+      it('should follow the keyboard down', () => {
+        keyboardSlideOut();
+        expect(ionFooterElement).toHaveComputedStyle({
+          transform: 'none',
+        });
+      });
+    });
+
+    describe('when snapToKeyboard is false', () => {
+      it('should not follow the keyboard up', () => {
+        spectator.setHostInput('snapToKeyboard', false);
+        keyboardSlideIn();
+        expect(ionFooterElement).toHaveComputedStyle({
+          transform: 'none',
```

Comment:
when testing a change of state based on an action, it is a good idea to verify the value wasn't the same before the test